react-renderer/src/Optimizer.ts

```diff
@@ -71,9 +65,9 @@ export const constrDict = {
   },
 };
 
-export const center = (props: any): Tensor1D =>
-  tf.stack([props.x.contents, props.y.contents]) as Tensor1D; // HACK: need to annotate the types of x and y to be Tensor
-export const dist = (p1: Tensor1D, p2: Tensor1D) => p1.sub(p2).norm(); // NOTE: or tf.squaredDifference
+export const center = (props: any): Tensor =>
```

Comment:
What's the relationship between Tensor and Variable types in typescript? Is it doing "permissive subtyping" (any variable can be used as a tensor) or would it ever coerce the variable type either way?

Reply:
Variable extends Tensor, so the latter thing you said